examples/generator/bernstein_vazirani.py

```diff
@@ -1,189 +0,0 @@
-    """
-
-    def __init__(self, qubit_count: int = 8, pattern: int = 12):
-        """
-        :param qubit_count: number of qubits to use
-        :type qubit_count: int
-        :param pattern: integer representation of the bitstring pattern
-        :type pattern: int
-        """
-        self.builder = QirBuilder("Bernstein-Vazirani")
-        bitstring = self.int_as_bool_array(pattern, qubit_count)
-        Uf = self.parity_operation(bitstring)
-        self.parity_via_fourier_sampling(Uf, qubit_count)
-
-    def write_ir_file(self, file_path: str):
-        self.builder.write(file_path)
-
-    def get_ir_string(self) -> str:
-        return self.builder.get_ir_string()
-
-    def generate_ir_file(file_path: str, qubit_count=8, pattern=12):
-        instance = BernsteinVazirani(qubit_count, pattern)
-        instance.write(file_path)
-
-    def parity_operation_impl(self, pattern: List[bool],
-                              queryRegister: List[str], target: str) -> None:
-        """
-        To demonstrate the Bernstein–Vazirani algorithm, we define
-        a function which returns black-box operations (List[str] => None) of
-        the form
-
-            U_f |𝑥〉|𝑦〉 = |𝑥〉|𝑦 ⊕ 𝑓(𝑥)〉,
-
-        In particular, we define 𝑓 by providing the pattern 𝑟⃗. Thus, we can
-        easily assert that the pattern measured by the Bernstein–Vazirani
-        algorithm matches the pattern we used to define 𝑓.
-
-        We will typically only call this function by partially applying it from
-        within a matching function.
-
-        :param pattern: The bitstring 𝑟⃗ used to define the function 𝑓.
-        :type pattern: List[bool]
-        :param queryRegister: qubit ids to perform operations against
-        :type queryRegister: List[str]
-        :param target: name of the target qubit
-        :type target: str
-        """
-
-        if len(queryRegister) != len(pattern):
-            raise ValueError(
-                'Length of input register must be equal to the pattern length.'
-            )
-
-        for patternBit, controlQubit in zip(pattern, queryRegister):
-            if patternBit:
-                self.builder.cx(controlQubit, target)
-
-    def parity_operation(self, pattern: List[bool]) -> ParityFunction:
-        """
-        Given a bitstring 𝑟⃗ = (r₀, …, rₙ₋₁), returns an operation implementing
-        a unitary 𝑈 that acts on 𝑛 + 1 qubit ids as
-            𝑈 |𝑥〉|𝑦〉 = |𝑥〉|𝑦 ⊕ 𝑓(𝑥)〉,
-        where 𝑓(𝑥) = Σᵢ 𝑥ᵢ 𝑟ᵢ mod 2.
-
-        :param pattern: The bitstring 𝑟⃗ used to define the function 𝑓.
-        :type pattern: List[bool]
-        Returns:
-            An operation implementing 𝑈.
-        """
-        return lambda register, target: self.parity_operation_impl(
-            pattern,
-            register,
-            target
-        )
-
-    def parity_via_fourier_sampling(self, Uf: ParityFunction, n: int) -> None:
-        """
-        parity_via_fourier_sampling implements the Bernstein-Vazirani quantum
```

Comment:
Is this comment now lost? I think it makes sense to keep the description of the algorithm somewhere? Maybe make mock up language support comments and put it in the Bernstein-Vazirani.txt?

Reply:
It supports comments but not block comments. I'll defer figuring out how to support block comments for another time, but I added a link.